src/screens/ProfileScreen/elements/TextWithEmoji.tsx

```diff
@@ -4,12 +4,22 @@ import emojiRegex from 'emoji-regex';
 import {COLORS} from '../../../utils/theme';
 import {fonts} from '../../../utils/fonts';
 
-export function TextWithEmoji({text}: {text: string}) {
+export function TextWithEmoji({
+  text,
+  textStyle,
+  testId
+}: {
+  text: string;
+  textStyle: object;
+  testId: string;
+}) {
   return (
-    <Text style={styles.text}>
-      {text.split(' ').map((item: string, index: number) => {
+    <Text style={[styles.text, textStyle]} testID={testId}>
+      {text?.split(' ').map((item: string, index: number) => {
         return (
-          <Text key={index} style={item.match(emojiRegex()) ? styles.emoji : styles.text}>
+          <Text
+            key={index}
+            style={[item.match(emojiRegex()) ? styles.emoji : styles.text, textStyle]}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for applying styles to text elements has been adjusted. Now, the `textStyle` prop is applied to both the parent and child `Text` components. This could potentially lead to conflicts or unexpected behavior if the same style properties are defined in both `styles.text` and `textStyle`. Consider applying the `textStyle` only to the parent `Text` component, or ensure that the consumers of this component are aware of this behavior.

Also, using index as a key in React is generally not recommended because it doesn't uniquely identify your elements. If the order of items changes, it might cause issues with component rendering. If each `item` is guaranteed to be unique, you could use that as a key instead.

```diff
- <Text key={index} style={[item.match(emojiRegex()) ? styles.emoji : styles.text, textStyle]}>
+ <Text key={item} style={[item.match(emojiRegex()) ? styles.emoji : styles.text]}>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->